addons/bus/static/src/js/bus.js

```diff
@@ -1,130 +1,85 @@
-odoo.define('bus.bus', function (require) {
+odoo.define('bus.Longpolling', function (require) {
 "use strict";
 
-var local_storage = require('web.local_storage');
-var session = require('web.session');
-var Widget = require('web.Widget');
+var Bus = require('web.Bus');
+var ServicesMixin = require('web.ServicesMixin');
 
-var bus = {};
-var PARTNERS_PRESENCE_CHECK_PERIOD = 30000;  // don't check presence more than once every 30s
 
-var TAB_HEARTBEAT_PERIOD = 10000;  // 10 seconds
-var MASTER_TAB_HEARTBEAT_PERIOD = 1500;  // 1.5 second
+/**
+ * Event Longpolling bus used to bind events on the server long polling return
+ *
+ * trigger:
+ * - window_focus : when the window is focused
+ * - notification : when a notification is receive from the long polling
+ *
+ * @class Longpolling
+ */
+return Bus.extend(ServicesMixin, {
+    // constants
+    partnersPresenceCheckPeriod: 30000,  // don't check presence more than once every 30s
+    onErrorRetryDelay: 10000, // 10 seconds
+    pollRoute: '/longpolling/poll',
 
-bus.ERROR_DELAY = 10000;
+    // properties
+    _activated: false,
```

Comment:
All things considered, it's okay because "default" values are on prototype, and whenever there is a change of value, a new variable in the scope of the instance is created, shadowing the one in the prototype. In the end, this is not a problem.